internal/clickhouse/stats.go

```diff
@@ -34,6 +35,28 @@ type TableStat struct {
 // ColumnStats retrieves detailed statistics for each column of a specific table.
 func (c *Client) ColumnStats(ctx context.Context, database, table string) ([]TableColumnStat, error) {
 	// Query system.parts_columns for statistics on active parts.
+	engine,err := c.getEngine(ctx, database, table)
+	var actualTable string
+	if err !=nil {
+		fmt.Println(err)
+	}
+	switch engine{
+	case "Distributed":
+		var engine_full,err = c.getEngineTable(ctx, database,table)
+		if err !=nil {
+			fmt.Println(err)
+		}
+		pattern := `Distributed\(.*?,\s*'(.*?)'\s*,\s*'(.*?)'`
```

Comment:
use `client.getBaseTableInfo` to figure if it's Distributed engine. Remove the other duplicate methods.